actionmailer/lib/action_mailer/base.rb

```diff
@@ -801,47 +797,54 @@ def _raise_error
     def mail(headers = {}, &block)
       return @_message if @_mail_was_called && headers.blank? && !block
 
-      m = @_message
-
       # At the beginning, do not consider class default for content_type
       content_type = headers[:content_type]
 
-      # Call all the procs (if any)
-      default_values = {}
-      self.class.default.each do |k,v|
-        default_values[k] = v.is_a?(Proc) ? instance_eval(&v) : v
-      end
-
-      # Handle defaults
-      headers = headers.reverse_merge(default_values)
-      headers[:subject] ||= default_i18n_subject
+      headers = apply_defaults(headers)
 
       # Apply charset at the beginning so all fields are properly quoted
-      m.charset = charset = headers[:charset]
+      message.charset = charset = headers[:charset]
 
       # Set configure delivery behavior
       wrap_delivery_behavior!(headers.delete(:delivery_method), headers.delete(:delivery_method_options))
 
-      # Assign all headers except parts_order, content_type, body, template_name, and template_path
-      assignable = headers.except(:parts_order, :content_type, :body, :template_name, :template_path)
-      assignable.each { |k, v| m[k] = v }
+      assign_headers_to_message(headers)
```

Comment:
That's what I'd like to change for all such methods, because `message` is already stored as an instance variable, and I see no point in passing it around as an argument. But I understand this will change signatures of protected methods, possibly breaking inherited classes in users' projects and other gems.

So there, I changed it for consistency: https://github.com/rails/rails/commit/2a55a6d2e518662d0f213cfe8c81b2eea66b5323